pkg/solana/chainreader/chain_reader.go

```diff
@@ -145,11 +163,11 @@ func (s *SolanaChainReaderService) GetLatestValue(ctx context.Context, readIdent
 			}
 
 			wg.Add(1)
-			go func(ctx context.Context, rb readBinding, res *loadedResult) {
+			go func(ctx context.Context, rb readBinding, res *loadedResult, address string) {
 				defer wg.Done()
 
-				rb.PreLoad(ctx, res)
-			}(localCtx, binding, results[idx])
+				rb.PreLoad(ctx, address, res)
```

Comment:
Is Preload similar to evm contract read? Why not call ReadAsync?

Reply:
For the reader, `PreLoad` does do an async read, but it stores the value in the reader and must be followed by `GetLatestValue`. It allows multiple readers to be read async, but in order.